pkg/block/s3/client_cache.go

```diff
@@ -142,7 +142,7 @@ func (c *ClientCache) cachedClientByRegion(region string) (*s3.Client, bool) {
 
 func (c *ClientCache) refreshBucketRegion(ctx context.Context, bucket string) string {
 	region, err := c.s3RegionGetter(ctx, bucket)
-	if err != nil {
+	if err != nil || region == "" {
```

Comment:
If we the upgrade requires _any_  code change - we must explain why first to make sure we understand the change.